src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureUnitTests.cs

```diff
@@ -890,5 +891,655 @@ static void Main(string[] args)
             test.ExpectedDiagnostics.Add(VerifyCS.Diagnostic(DetectPreviewFeatureAnalyzer.GeneralPreviewFeatureAttributeRule).WithLocation(0).WithArguments("SampleEvent"));
             await test.RunAsync();
         }
+
+        [Fact]
+        public async Task TestTypeOf()
+        {
+            var csInput = @" 
+using System.Runtime.Versioning; using System;
+namespace Preview_Feature_Scratch
+{" +
+@"
+
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Console.WriteLine({|#0:typeof(IFoo)|});
+        }
+    }
+    
+    [RequiresPreviewFeatures]
+    interface IFoo { }
+}";
+
+            var test = TestCS(csInput);
+            test.ExpectedDiagnostics.Add(VerifyCS.Diagnostic(DetectPreviewFeatureAnalyzer.GeneralPreviewFeatureAttributeRule).WithLocation(0).WithArguments("IFoo"));
+            await test.RunAsync();
+        }
+
+        [Fact]
+        public async Task TestDerivedInterface()
+        {
+            var csInput = @" 
+using System.Runtime.Versioning; using System;
+namespace Preview_Feature_Scratch
+{" +
+@"
+
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Program program = new Program();
+        }
+    }
+
+    interface {|#0:IZoo|} : IFoo
+    {
+    }
+
+    [RequiresPreviewFeatures]
+    interface IFoo
+    {
+        void Bar();
+    }
+}";
+
+            var test = TestCS(csInput);
+            test.ExpectedDiagnostics.Add(VerifyCS.Diagnostic(DetectPreviewFeatureAnalyzer.GeneralPreviewFeatureAttributeRule).WithLocation(0).WithArguments("IZoo"));
+            await test.RunAsync();
+
+        }
+
+        [Fact]
+        public async Task TestGenericClassWithoutPreviewInterface()
+        {
+            var csInput = @" 
+using System.Runtime.Versioning; using System;
+namespace Preview_Feature_Scratch
+{" +
+@"
+
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            A<Foo> aFooInstance = {|#0:new A<Foo>()|}; // Can we have a property?
+        }
+    }
+
+class A<T> where T : IFoo, new()
```

Comment:
Why no warning here?

Reply:
Because `IFoo` is not preview. Only `Foo` is

Reply:
thanks, I misread it. ignore me